src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -431,6 +434,13 @@ static Var *
 FindUnionAllVar(PlannerInfo *root, List *translatedVars, Oid relationOid,
 				Index relationRteIndex, Index *partitionKeyIndex)
 {
+	if (!IsCitusTableType(relationOid, STRICTLY_PARTITIONED_DISTRIBUTED_TABLE))
+	{
+		/* we only care about hash and range partitioned tables */
+		*partitionKeyIndex = 0;
+		return NULL;
```

Comment:
would that make sense to add a single test with reference table to trigger this line?

Reply:
yes, added one test with local table and one with reference